src/tests/JIT/HardwareIntrinsics/Arm/Shared/SveStoreAndZipTest.template

```diff
@@ -0,0 +1,336 @@
+            }
+        }
+
+
+        private struct TestStruct
+        {
+            public {Op1VectorType}<{Op1BaseType}> _fld1;
+            public {Op1VectorType}<{Op1BaseType}> _storeMask;
+            private GCHandle _outHandle;
+
+            public static TestStruct Create()
+            {
+                var testStruct = new TestStruct();
+                var rnd        = new Random();
+
+                for (var i = 0; i < Op1ElementCount; i++) { _data1[i] = {NextValueOp1}; }
+                Unsafe.CopyBlockUnaligned(ref Unsafe.As<{Op1VectorType}<{Op1BaseType}>, byte>(ref testStruct._fld1), ref Unsafe.As<{Op1BaseType}, byte>(ref _data1[0]), (uint)Unsafe.SizeOf<{Op1VectorType}<{Op1BaseType}>>());
+                for (var i = 0; i < MaskElementCount; i++) { _maskData[i] = ({Op1BaseType})1; }
+                for (var i = 0; i < MaskElementCount / 2; i++) { _maskData[rnd.Next(MaskElementCount)] = ({Op1BaseType})0; }
+                Unsafe.CopyBlockUnaligned(ref Unsafe.As<{Op1VectorType}<{Op1BaseType}>, byte>(ref testStruct._storeMask), ref Unsafe.As<{Op1BaseType}, byte>(ref _maskData[0]), (uint)Unsafe.SizeOf<{Op1VectorType}<{Op1BaseType}>>());
+
+                ulong alignment  = (ulong)LargestVectorSize;
+                byte[] _outArray = new byte[alignment * 2];
+                testStruct._outHandle = GCHandle.Alloc(_outArray, GCHandleType.Pinned);
+
+                return testStruct;
+            }
+
+            public void* _outArrayPtr => DataTable.Align((byte*)(_outHandle.AddrOfPinnedObject().ToPointer()), (ulong)LargestVectorSize);
+
+            public void Dispose()
+            {
+                _outHandle.Free();
+            }
+
+            public void RunStructFldScenario(StoreTest__{TestName} testClass)
+            {
+                Sve.{Method}(_storeMask, ({Op1BaseType}*) _outArrayPtr, _fld1);
+
+                testClass.ValidateResult(_fld1, _outArrayPtr, _storeMask);
+            }
+        }
+
+        private static readonly int LargestVectorSize = {LargestVectorSize};
+
+        private static readonly int Op1ElementCount  = Unsafe.SizeOf<{Op1VectorType}<{Op1BaseType}>>() / sizeof({Op1BaseType});
+        private static readonly int RetElementCount  = Unsafe.SizeOf<{RetVectorType}<{RetBaseType}>>() / sizeof({RetBaseType});
+        private static readonly int MaskElementCount = Unsafe.SizeOf<{Op1VectorType}<{Op1BaseType}>>() / sizeof({Op1BaseType});
+
+        private static {Op1BaseType}[] _data1    = new {Op1BaseType}[Op1ElementCount];
+        private static {Op1BaseType}[] _maskData = new {Op1BaseType}[MaskElementCount];
+
+        private DataTable _dataTable;
+
+        public StoreTest__{TestName}()
+        {
+            Succeeded = true;
+            var rnd   = new Random();
+
+            for (var i = 0; i < Op1ElementCount; i++) { _data1[i] = {NextValueOp1}; }
+            for (var i = 0; i < MaskElementCount; i++) { _maskData[i] = ({Op1BaseType})1; }
```

Comment:
can we write it simpler?

```c#
for (var i = 0; i < MaskElementCount; i++) {  _maskData[i] = ({Op1BaseType})({NextValueOp1} % 2); }
```

Reply:
please send a follow up PR. I will merge this one because it is a minor rewrite.

Reply:
Cool, I'll add this in the `StoreNarrowing` PR.